metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -433,6 +433,58 @@ def _make_gcp_logging_client(
         else:
             return [GCPLoggingClient(**client_options)]
 
+    def generate_profile_candidates(
+        self,
+        inspector: Inspector,
+        threshold_time: Optional[datetime.datetime],
+        schema: str,
+    ) -> Optional[List[str]]:
+        row_condition = (
+            f"row_count<{self.config.profiling.profile_table_row_limit} and "
```

Comment:
The first half of this method (line 442 - 471)  is worth refactoring and writing a unit test for. 
Depending on various settings, what are the query strings generated?